src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,30 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> eventListenerList = new CopyOnWriteArrayList<>();
+    private final ExecutorService executor = Executors.newFixedThreadPool(3);
 
+    public void registerListener(EventListener listener) {
+        eventListenerList.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        eventListenerList.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : eventListenerList) {
+            executor.submit(() -> listener.onEvent(event));
+        }
```

Comment:
How about making it a single function call?
`executor.submit(() -> listeners.forEach(listener -> listener.onEvent(event)));`